### DISCUSSION TITLE
Avalonia the client hub connection is working fine on the desktop but not working for browser

### DISCUSSION THREAD

**Discussion by IftikharK** [AvaloniaUI/Avalonia#14459] [General]
i am new to Avalonia + SignalR. I am trying to create a simple Real-Time Chat cross platform application (desktop, browser, android, ios) using Avalonia 11.0.6 and SignalR . It work fine when I run for desktop application, but it is not working in browser application.  I feel connection is not established with SignalR when trying to start await connection.StartAsync();

Browser app launchSettings.json have  "applicationUrl": "https://localhost:5001;http://localhost:5000",

![ChatHub](https://github.com/AvaloniaUI/Avalonia/assets/47390767/5598fc08-3d04-4132-98b6-cf01c796ca26)

![MainView axaml cs](https://github.com/AvaloniaUI/Avalonia/assets/47390767/2b99aef2-6729-4e75-adcc-bfa02601803c)

![SignalR_Program cs](https://github.com/AvaloniaUI/Avalonia/assets/47390767/0e844a79-12c5-420d-a883-285ec82cd15e)

**Comment by thevortexcloud** [comment]
>  feel connection is not established with SignalR when trying to start await connection.StartAsync();

That's possible. Web browsers are fairly sandboxed. Although SignalR should be using web sockets which has far less restrictions on it. So I am not too sure what's going on here. Have you tested if Blazor works? It officially uses SignalR by default.

**Comment by timunie** [comment]
Please try to use nightly builds for async stuff and Browser

**Reply by IftikharK** [reply]
@timunie Sorry, I didn't understand. What do you mean by 'nightly builds'?

**Reply by timunie** [reply]
Nightly are preview builds of upcoming 11.1. they have own nuget feed https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed

 even update to 11.0.7 should help.

**Reply by IftikharK** [reply]
@timunie Unfortunately, my attempts with Avalonia 11.0.7 and then 11.1.999-cibuild0044396-beta were both unsuccessful. 
In the latest master branch, browser platform completely stopped working. It is stuck on the 'Powered by Avalonia' html welcome page, console is displaying the following warning.
![warn](https://github.com/AvaloniaUI/Avalonia/assets/47390767/7940eb60-9403-4ff2-b977-3afd70393bb0)

**Reply by thevortexcloud** [reply]
There were some changes for async webasm recently from memory. You need to use the latest templates and/or .NET 8 for it to work properly.

https://github.com/AvaloniaUI/Avalonia/issues/14022
